### PR TITLE
docs: revamp CoreOS getting started guides

### PR DESCRIPTION
**Pull request by robszumski**
- Split the guides into two sections: official and community
- Added descriptions to more clearly articulate the differences between them

Fixes https://github.com/kubernetes/kubernetes/issues/14205, https://github.com/kubernetes/kubernetes/issues/9178, and https://github.com/kubernetes/kubernetes/issues/14032

CLA should be pre-signed by CoreOS, Inc.

### CODE REVIEW

File: docs/getting-started-guides/coreos.md [hunk 1]
```diff
@@ -31,17 +31,73 @@ Documentation for other releases can be found at
...
+
+[**Resizable multi-node cluster on Azure with Weave**](coreos/azure/README.md)
+
+Guide to running an HA etcd cluster with a single master on Azure. Uses the Azure node.js CLI to resize the cluster.
+
+<hr/>
+
+[**Multi-node cluster using cloud-config, coreos and VMware ESXi**](https://github.com/xavierbaude/VMware-coreos-multi-nodes-Kubernetes)
+
+Configure a singe master, single worker cluster on VMware EXSi.
```
Comment (jonboulle) [inline docs/getting-started-guides/coreos.md]: ESXi

File: docs/getting-started-guides/coreos.md [hunk 2]
```diff
@@ -31,17 +31,73 @@ Documentation for other releases can be found at
...
+
+[**Multi-node cluster using cloud-config and Vagrant**](https://github.com/pires/kubernetes-vagrant-coreos-cluster/blob/master/README.md)
+
+Configure a single-master, multi-worker cluster locally, running on your choice of hypervisor: VirtualBox, Parallels, or VMware
+
+<hr/>
+
+[**Multi-node cluster with Vagrant and fleet units using a small OS X App**](https://github.com/rimusz/coreos-osx-gui-kubernetes-cluster/blob/master/README.md)
+
+Guide to running a single master, multi-worker cluster controlled by an OS X menubar application. Uses Vagrant under the hood.
```
Comment (jonboulle) [inline docs/getting-started-guides/coreos.md]: single-master

File: docs/getting-started-guides/coreos.md [hunk 3]
```diff
@@ -31,17 +31,73 @@ Documentation for other releases can be found at
...
+
+[**Multi-node cluster using cloud-config and Weave on Vagrant**](https://github.com/errordeveloper/weave-demos/blob/master/poseidon/README.md)
+
+Configure a Vagrant-based cluster of 3 machines with networking provided by Weave.
+
+<hr/>
+
+[**Multi-node cluster using cloud-config and Vagrant**](https://github.com/pires/kubernetes-vagrant-coreos-cluster/blob/master/README.md)
+
+Configure a single-master, multi-worker cluster locally, running on your choice of hypervisor: VirtualBox, Parallels, or VMware
```
Comment (jonboulle) [inline docs/getting-started-guides/coreos.md]: "single master" for consistency with elsewhere, or vice versa, no strong feeling either way

File: docs/getting-started-guides/coreos.md [hunk 4]
```diff
@@ -31,17 +31,73 @@ Documentation for other releases can be found at
...
+
+<hr/>
+
+[**Full Step by Step Guide**](https://coreos.com/kubernetes/docs/latest/getting-started.html)
+
+A generic guide to setting up an HA cluster on any cloud or bare metal, with full TLS. Repeat the master or worker steps to configure more machines of that role.
+
+### Community Guides
+
+These guides are maintained by community members and cover specific platforms, use-cases and experiment with different ways of configuring Kubernetes on CoreOS.
```
Comment (jonboulle) [inline docs/getting-started-guides/coreos.md]: ...by community members, cover specific platforms and use cases, and experiment with ...

File: docs/getting-started-guides/coreos.md [hunk 5]
```diff
@@ -31,17 +31,73 @@ Documentation for other releases can be found at
...
+
+[**Vagrant Multi-Node**](https://coreos.com/kubernetes/docs/latest/kubernetes-on-vagrant.html)
+
+Guide to setting up a multi-node cluster on Vagrant. The deployer can independently configure the number of etcd nodes, master nodes, and worker nodes to bring up a fully HA control plane.
+
+<hr/>
+
+[**Vagrant Single-Node**](https://coreos.com/kubernetes/docs/latest/kubernetes-on-vagrant-single.html)
+
+Quickest way to set up a Kubernetes development environment locally. As easy as `git clone`, `vagrant up` and configuring `kubectl`.
```
Comment (jonboulle) [inline docs/getting-started-guides/coreos.md]: Quickest -> The quickest

File: docs/getting-started-guides/coreos.md (hunk 1 above)
Comment (jonboulle) [inline docs/getting-started-guides/coreos.md]: er, and **single**

**Comment by robszumski** [comment]
Thanks for the review @jonboulle, updated.

File: docs/getting-started-guides/coreos.md [hunk 6]
```diff
@@ -31,17 +31,73 @@ Documentation for other releases can be found at
...
+
+<hr/>
+
+[**Resizable multi-node cluster on Azure with Weave**](coreos/azure/README.md)
+
+Guide to running an HA etcd cluster with a single master on Azure. Uses the Azure node.js CLI to resize the cluster.
+
+<hr/>
+
+[**Multi-node cluster using cloud-config, coreos and VMware ESXi**](https://github.com/xavierbaude/VMware-coreos-multi-nodes-Kubernetes)
```
Comment (jonboulle) [inline docs/getting-started-guides/coreos.md]: coreos -> CoreOS

**Comment by jonboulle** [comment]
@robszumski sorry, one more thing (we should at least get our own brand right ;-)

**Comment by jonboulle** [comment]
LGTM, fwiw.

**Comment by pires** [comment]
There's no point, at least on my side, in maintaining community-supported documentation. So, it can be removed.

**Pull request merged**